# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ObjectGroup.h
# Commit: ec72f104edbf
# Full Hash: ec72f104edbfe88e22e2ebd778cc7890c21fd2ce
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 9 - Move proto from ObjectGroup to BaseShape. r=jonco
#   
#   The ObjectGroup still stores the proto as well but we don't really use it anymore.
#   This is simpler than removing the group's proto now because that would result in
#   "empty" groups and the group table then doesn't make sense anymore.
# ==============================================================================

diff -r cad616508f73 -r ec72f104edbf js/src/vm/ObjectGroup.h
--- a/js/src/vm/ObjectGroup.h	Sat Mar 06 01:05:22 2021 +0000
+++ b/js/src/vm/ObjectGroup.h	Sat Mar 06 01:05:22 2021 +0000
@@ -55,10 +55,6 @@
   // END OF PROPERTIES
 
  private:
-  static inline uint32_t offsetOfProto() {
-    return offsetof(ObjectGroup, proto_);
-  }
-
   friend class gc::GCRuntime;
 
   // See JSObject::offsetOfGroup() comment.
@@ -67,11 +63,11 @@
  public:
   inline explicit ObjectGroup(TaggedProto proto);
 
-  const GCPtr<TaggedProto>& proto() const { return proto_; }
+  const GCPtr<TaggedProto>& protoDeprecated() const { return proto_; }
 
-  GCPtr<TaggedProto>& proto() { return proto_; }
+  GCPtr<TaggedProto>& protoDeprecated() { return proto_; }
 
-  void setProtoUnchecked(TaggedProto proto);
+  void setProtoUncheckedDeprecated(TaggedProto proto);
 
   /* Helpers */
 